MC-783 Documentation

hippietilley opened the pull request:
MC-783 Documentation

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/md2korg/mcerebrum-datakitapi/20)
<!-- Reviewable:end -->

hippietilley committed:
MC 783 - Documented DateTime

hippietilley committed:
MC 783 - Documented Application and ApplicationBuilder

hippietilley committed:
MC 783 - Documented DataType

hippietilley committed:
MC 783 - Started datatype documentation

hippietilley committed:
MC 783 - Documented DataTypeBoolean

hippietilley committed:
MC 783 - Documented DataTypeBooleanArray

hippietilley committed:
MC 783 - Documented DataTypeByte

hippietilley committed:
MC 783 - Documented DataTypeByteArray

hippietilley committed:
MC 783 - Documented DataTypeDouble

hippietilley committed:
MC 783 - Fixed typo

hippietilley committed:
MC 783 - Documented DataTypeDoubleArray

hippietilley committed:
MC 783 - Documented DataTypeFloat

hippietilley committed:
MC 783 - Documented DateTime

hippietilley committed:
MC 783 - Documented DataTypeInt

hippietilley committed:
MC 783 - Documented DataTypeIntArray

hippietilley committed:
MC 783 - Documented DataTypeJSONObject

hippietilley committed:
MC 783 - Fixed typo

hippietilley committed:
MC 783 - Documented DataTypeJSONObjectArray

hippietilley committed:
MC 783 - Documented DataTypeLong

hippietilley committed:
MC 783 - Documented DataTypeLongArray

hippietilley committed:
MC 783 - Documented DataTypeString

hippietilley committed:
MC 783 - Documented DataTypeStringArray

hippietilley committed:
MC 783 - Documented RowObject

hippietilley committed:
MC 783 - Documented DataKitException, DataKitNotFoundException, and Status

hippietilley committed:
MC 783 - Documented METADATA

hippietilley committed:
MC 783 - Documented AbstractObject

hippietilley committed:
MC 783 - Documented AbstractObjectBuilder

hippietilley committed:
MC 783 - Added class doc for ApplicationBuilder

hippietilley committed:
MC 783 - Documented PlatformBuilder

hippietilley committed:
MC 783 - Documented PlatformId

hippietilley committed:
MC 783 - Documented PlatformType

hippietilley committed:
MC 783 - Documented Platform

hippietilley committed:
MC 783 - Documented PlatformApp and PlatformAppBuilder

hippietilley committed:
MC 783 - Documented DataSource folder and Constants

hippietilley committed:
MC 783 - Added copyright to the rest of the files

twhnat commented:
@hippietilley Please use MC-783 instead of MC 783 to tag this issue.  It must match precisely with what the Issue ID is in the system.

hippietilley committed:
MC-783 - Documented DataSourceType

hippietilley committed:
MC-783 - Finished DataSourceType

hippietilley committed:
MC-783 - Documented MessageType

hippietilley committed:
MC-783 - Started DataKitAPI documentation

twhnat commented:
Reviewed 46 of 49 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 54 unresolved discussions.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/Constants.java, line 57 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bj6YMGYNqSZHi4CK2:-L4bj6YMGYNqSZHi4CK3:bdykr8n) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/Constants.java#L57)):*
> ```Java
>      * </p>
>      */
>     public static final String RC_DSID = "ds_id";
> ```

@monowar What is the meaning of RC?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 53 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjDlzOZ3EBThzRqZM:-L4bjDlzOZ3EBThzRqZN:b-g211ig) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L53)):*
> ```Java
>  *
>  */
> public class DataKitAPI {
> ```

Needs a comment

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 55 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjH63kmRFGCs6KE6E:-L4bjH63kmRFGCs6KE6F:b8a2wp9) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L55)):*
> ```Java
> public class DataKitAPI {
>     private static final String TAG = DataKitAPI.class.getSimpleName();
>     private static final int BUFFER_SIZE = 1<<13; //8 KB
> ```

Why 8KB?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 65 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjL2NvmLXEC58h-lb:-L4bjL2NvmLXEC58h-lc:b-t31w8a) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L65)):*
> ```Java
>     Handler handler;
> 
>     /** Milliseconds between high frequency syncing
> ```

Maximum time in milliseconds to wait when transmitting high-frequency data

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 77 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjXlzzcmNumUQo6aV:-L4bjXlzzcmNumUQo6aW:b-9xynyp) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L77)):*
> ```Java
>     HashMap<Integer, HFBuffer> hmHFBuffer;
> 
>     /** Embedded class for creating <code>HFBuffer</code> objects. */
> ```

... creating high-frequency ...

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 145 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjpkhHc5jT6DypeVG:-L4bjpkhHc5jT6DypeVH:blbsm49) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L145)):*
> ```Java
>     }
> 
>     /**
> ```

Needs method descriptor for connect

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 184 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjvEomOtnijceAHA9:-L4bjvEomOtnijceAHAA:b-8of0xg) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L184)):*
> ```Java
>     }
> 
>     /**
> ```

Need method definition comment

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 203 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEE_psVkQheJ5KoXT:-L4dEE_psVkQheJ5KoXU:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L203)):*
> ```Java
>         }
>     }
>     public synchronized void setSummary(DataSourceClient dataSourceClient, DataType dataType) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 213 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEG8dA8xYwc884X_Q:-L4dEG8dA8xYwc884X_R:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L213)):*
> ```Java
>     }
> 
>     public synchronized void insert(DataSourceClient dataSourceClient, DataType[] dataTypes) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 220 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEGv4Mku8tfxf_yMp:-L4dEGv4Mku8tfxf_yMq:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L220)):*
> ```Java
>         else dataKitAPIExecute.insert(dataSourceClient, dataTypes);
>     }
>     public synchronized void insertHighFrequency(final DataSourceClient dataSourceClient, final DataTypeDoubleArray[] dataType) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 231 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEHl_H4Glbbw0T1Ei:-L4dEHl_H4Glbbw0T1Ej:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L231)):*
> ```Java
>     }
> 
>     public synchronized void insertHighFrequency(final DataSourceClient dataSourceClient, final DataTypeDoubleArray dataType) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 240 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEKU6te4Q8stCQHxo:-L4dEKU6te4Q8stCQHxp:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L240)):*
> ```Java
>     }
> 
>     synchronized void addToBuffer(int ds_id, DataTypeDoubleArray dataTypeDoubleArray) {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 256 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dELG4StB1E426Tv4E:-L4dELG4StB1E426Tv4F:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L256)):*
> ```Java
>     }
> 
>     synchronized void syncHFData(int ds_id) {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 271 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEM4-cIMMIBchVdde:-L4dEM4-cIMMIBchVddf:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L271)):*
> ```Java
>     }
> 
>     synchronized void syncHFDataAll() {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 276 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEMlYrB4A8y7BGAM7:-L4dEMlYrB4A8y7BGAM8:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L276)):*
> ```Java
>     }
> 
>     public synchronized DataSourceClient register(final DataSourceBuilder dataSourceBuilder) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 287 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dENSh4yNNbHrrK0FG:-L4dENSh4yNNbHrrK0FH:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L287)):*
> ```Java
>     }
> 
>     public synchronized Status unregister(final DataSourceClient dataSourceClient) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 298 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEOBRVOmkk2Q_DKZ5:-L4dEOBRVOmkk2Q_DKZ6:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L298)):*
> ```Java
>     }
> 
>     public synchronized ArrayList<DataType> query(final DataSourceClient dataSourceClient, final int last_n_sample) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 309 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEOut4ueIW4l_W8hn:-L4dEOuuhbFycoxFFZB-:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L309)):*
> ```Java
>     }
> 
>     public synchronized ArrayList<DataType> query(DataSourceClient dataSourceClient, long starttimestamp, long endtimestamp) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 320 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEPpSV9prlydEbFQn:-L4dEPpSV9prlydEbFQo:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L320)):*
> ```Java
>     }
> 
>     public synchronized ArrayList<RowObject> queryFromPrimaryKey(DataSourceClient dataSourceClient, long lastSyncedKey, int limit) throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 331 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dER1ptjHVCVddXDfE:-L4dER1ptjHVCVddXDfF:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L331)):*
> ```Java
>     }
> 
>     public synchronized DataTypeLong querySize() throws DataKitException {
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 410 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEbex4hK61maKMJt-:-L4dEbex4hK61maKMJt0:b-4qrhqv) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L410)):*
> ```Java
> 
>         /**
>          *
> ```

Initiates the high-frequency data transmit proceedure

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java, line 88 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dEvWj_wpKD4ziZNGl:-L4dEvWj_wpKD4ziZNGm:b-lyle4q) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java#L88)):*
> ```Java
> 
>     /**
>      * TODO: Not used. Can remove?
> ```

@monowar Can empty default constructors be removed from the java code?  If so, this should be removed

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java, line 110 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dF8BRQhDaOzoq1YIV:-L4dF8BRQhDaOzoq1YIW:b-40otht) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java#L110)):*
> ```Java
> 
>     /**
>      * @return Always returns 0.
> ```

@monowar Is there a good reason for this?

@hippietilley Where else in the codebase is this called?  This may give an insight into this reason.  Give that this is an @override, there may be a reason.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeBoolean.java, line 55 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dFchshYgbNmnVvVjF:-L4dFchshYgbNmnVvVjG:bsiufbt) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeBoolean.java#L55)):*
> ```Java
> 
>     /**
>      * TODO: Not used. Can remove?
> ```

Yes, assuming that it still compiles.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeJSONObject.java, line 44 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dFnug0YXdcBwWQXE1:-L4dFnug0YXdcBwWQXE2:bz87j7g) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeJSONObject.java#L44)):*
> ```Java
>      * <code>Creator</code> for <code>DataTypeJSONObject</code> objects.
>      *
>      * TODO: Consider putting this block (lines 45 - 68) down to the bottom of the class to be consistant with the other classes.
> ```

Please do

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/RowObject.java, line 40 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dFvyBoTmDtqavgsWI:-L4dFvyBoTmDtqavgsWJ:brarzdx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/RowObject.java#L40)):*
> ```Java
> 
> /**
>  *
> ```

Comment?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/messagehandler/OnConnectionFailedListener.java, line 32 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dGNRf34NMoD4kWxhv:-L4dGNRf34NMoD4kWxhw:b-qxz00x) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/messagehandler/OnConnectionFailedListener.java#L32)):*
> ```Java
> import org.md2k.datakitapi.status.Status;
> 
> public interface OnConnectionFailedListener {
> ```

Take a look at this comment https://stackoverflow.com/a/9234626 for the motivation behind interfaces.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/AbstractObject.java, line 57 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dHFuGBoksgGbp-dPs:-L4dHFuGBoksgGbp-dPt:b-m6xpkr) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/AbstractObject.java#L57)):*
> ```Java
>      * Constructor
>      *
>      * TODO: Does nothing. Can remove?
> ```

Empty constructors are required if this object can be created without parameters elsewhere in the code.  Check for places where it is called.  You can use IntelliJ to help identify this.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/METADATA.java, line 83 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dHbiKhpzQYjLbVc3e:-L4dHbiKhpzQYjLbVc3f:b-14qu48) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/METADATA.java#L83)):*
> ```Java
>     public static final String DEVICE_ID="DEVICE_ID";
> 
>     /** TODO: What's this? */
> ```

There is a notion of "descriptive statistics" (e.g. mean, median, 20th percentile, etc...) and this is a designator for it.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/application/Application.java, line 40 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dIEyXqPaq7k8z0-Wo:-L4dIEyXqPaq7k8z0-Wp:b-bif7kx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/application/Application.java#L40)):*
> ```Java
>  * This class creates application objects.
>  *
>  * TODO: Learn about json stuff
> ```

There are java annotations (https://en.wikipedia.org/wiki/Java_annotation)

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java, line 49 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dIVEsjVYrw6zV9R7Q:-L4dIVEsjVYrw6zV9R7R:bja1lmw) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java#L49)):*
> ```Java
>     public Platform platform = null;
> 
>     /**  */
> ```

Platform application

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java, line 53 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dIY0x6kxEITNGKrBf:-L4dIY0x6kxEITNGKrBg:b-5eb13a) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java#L53)):*
> ```Java
> 
>     /**  */
>     public Application application = null;
> ```

Android Application

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java, line 58 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dIbKAgNHD7x293MDf:-L4dIbKAgNHD7x293MDg:bhjn0zd) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceBuilder.java#L58)):*
> ```Java
>     public boolean persistent = true;
> 
>     /**  */
> ```

This provides a list of data descriptors as sets of tuples which describe the individual values of the datapoints.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 56 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJDGUDhba8wMyIux2:-L4dJDGUDhba8wMyIux3:b-mt6506) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L56)):*
> ```Java
>     public static final String LOCATION = "LOCATION";
> 
>     /** Geofence sensor <p><code>"GEOFENCE"</code></p> */
> ```

Remove the word sensor

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 74 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJMgKFkHkrWHM1yNz:-L4dJMgLGqNM7ysEqg3u:b-k0zpgo) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L74)):*
> ```Java
>     public static final String PACE = "PACE";
> 
>     /** Motion type sensor <p><code>"MOTION_TYPE"</code></p> */
> ```

Motion type inference

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 84 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJU13cXq3oUhUXRA7:-L4dJU13cXq3oUhUXRA8:b-isoscx) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L84)):*
> ```Java
> 
>     /** Calorie burn sensor <p><code>"CALORIE_BURN"</code></p>  */
>     public static final String CALORIE_BURN = "CALORIE_BURN";
> ```

While this is a typo, please revert it due to the potential breaking of the code.

@monowar Is this still used?

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 104 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJid2WzeX9JH7v7n_:-L4dJid2WzeX9JH7v7na:brsls8q) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L104)):*
> ```Java
>     public static final String AIR_PRESSURE = "AIR_PRESSURE";
> 
>     /** Respiration rate sensor <p><code>"RR_INTERVAL"</code></p> */
> ```

R-R Intervals (https://en.wikipedia.org/wiki/Heart_rate_variability)

Heart rate variability

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 114 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJu12B0ESfPqXa6I0:-L4dJu12B0ESfPqXa6I1:b-t9b0xq) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L114)):*
> ```Java
>     public static final String SKIN_TEMPERATURE = "SKIN_TEMPERATURE";
> 
>     /** Battery usage sensor <p><code>"BATTERY"</code></p> */
> ```

-sensor

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 117 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJtPSb76gfspWx211:-L4dJtPSb76gfspWx212:b-t9b0xq) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L117)):*
> ```Java
>     public static final String BATTERY = "BATTERY";
> 
>     /** CPU usage sensor <p><code>"CPU"</code></p> */
> ```

-sensor

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 120 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJsP82TtVvmh7JVGo:-L4dJsP82TtVvmh7JVGp:b-t9b0xq) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L120)):*
> ```Java
>     public static final String CPU = "CPU";
> 
>     /** Memory usage sensor <p><code>"MEMORY"</code></p> */
> ```

-sensor

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 123 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJwcMMuq1brpokZqD:-L4dJwcNZfDEyMTvs8DO:b-i8fmn4) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L123)):*
> ```Java
>     public static final String MEMORY = "MEMORY";
> 
>     /** Autosense sensor <p><code>"AUTOSENSE"</code></p> */
> ```

--sensor +device 

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 144 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dK1McmHk8N_hUfV9M:-L4dK1MdF99XzrPsmdSk:b-io09j7) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L144)):*
> ```Java
>     public static final String GYROSCOPE_Z = "GYROSCOPE_Z";
> 
>     /** TODO: Not sure what sensor this is... <p><code>"EMA"</code></p> */
> ```

Ecological Momentary Assessment

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 165 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dK7YmHdPut-J-XW5P:-L4dK7YmHdPut-J-XW5Q:b8reqmy) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L165)):*
> ```Java
>     public static final String USER_INFO = "USER_INFO";
> 
>     /** Sleep sensor <p><code>"SLEEP"</code></p> */
> ```

-sensor +time

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 168 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dKA4uK7_k68jdtp87:-L4dKA4uK7_k68jdtp88:b8reqmy) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L168)):*
> ```Java
>     public static final String SLEEP = "SLEEP";
> 
>     /** Wakeup sensor <p><code>"WAKEUP"</code></p> */
> ```

-sensor +time

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 243 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dKX7U5T_lAb6Bq3mj:-L4dKX7U5T_lAb6Bq3mk:bx5vihp) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L243)):*
> ```Java
> 
> 
>     /** Request of a smoking episode <p><code>"PUFFMARKER_SMOKING_EPISODE"</code></p> */
> ```

You messed up this string, check the original

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 246 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dKhOlf6Ukwqe8QywF:-L4dKhOlf6Ukwqe8QywG:bx5vihp) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L246)):*
> ```Java
>     public static final String NOTIFICATION_REQUEST = "PUFFMARKER_SMOKING_EPISODE";
> 
>     /** Acknowledgement of a smoking episode <p><code>"PUFFMARKER_SMOKING_EPISODE"</code></p> */
> ```

You messed up this string, check the original

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 38 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLLpIivsIt0kO-ZZp:-L4dLLpIivsIt0kO-ZZq:b-p41gzq) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L38)):*
> ```Java
>  *     Available types are:
>  *     <ul>
>  *         <li>Chest (Auto sensing)</li>
> ```

AutoSense

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 39 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLIPUn_2Cj1JsLx8g:-L4dLIPUn_2Cj1JsLx8h:b-e2pa44) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L39)):*
> ```Java
>  *     <ul>
>  *         <li>Chest (Auto sensing)</li>
>  *         <li>Low energy Bluetooth (Auto sensing)</li>
> ```

AutoSense Bluetooth Low Energy

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 40 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLGwSs7x5l4attFwP:-L4dLGwSs7x5l4attFwQ:b-35h6if) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L40)):*
> ```Java
>  *         <li>Chest (Auto sensing)</li>
>  *         <li>Low energy Bluetooth (Auto sensing)</li>
>  *         <li>Wrist (Auto sensing)</li>
> ```

AutoSense Wrist

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 45 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dL5QIwqi5p0GIx2Vr:-L4dL5QIwqi5p0GIx2Vs:bnry28d) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L45)):*
> ```Java
>  *         <li>Omron Blood Pressure</li>
>  *         <li>Omron Weight Scale</li>
>  *         <li>Easysense TODO: Does this refer to the Philips Easysense or something else? </li>
> ```

No, This is a custom sensor called EasySense

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 46 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLAu3RRSP5H13_WAZ:-L4dLAu3RRSP5H13_WA_:b-6kicod) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L46)):*
> ```Java
>  *         <li>Omron Weight Scale</li>
>  *         <li>Easysense TODO: Does this refer to the Philips Easysense or something else? </li>
>  *         <li>Motion sensor</li>
> ```

MotionSense

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 47 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLPQKwl3lHc2qhV4W:-L4dLPQKwl3lHc2qhV4X:bcehdca) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L47)):*
> ```Java
>  *         <li>Easysense TODO: Does this refer to the Philips Easysense or something else? </li>
>  *         <li>Motion sensor</li>
>  *         <li>OralB Toothbrush</li>
> ```

Oral-B Toothbrush

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 48 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLBoGK-YTFQvD6ZGW:-L4dLBoGK-YTFQvD6ZGX:br1vyxb) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L48)):*
> ```Java
>  *         <li>Motion sensor</li>
>  *         <li>OralB Toothbrush</li>
>  *         <li>Motion Sense with Heart Rate Variability</li>
> ```

MotionSense with Heart Rate Variability

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java, line 49 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dLR7n9S010gE6T0VN:-L4dLR7n9S010gE6T0VO:b-7qx1u9) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/platform/PlatformType.java#L49)):*
> ```Java
>  *         <li>OralB Toothbrush</li>
>  *         <li>Motion Sense with Heart Rate Variability</li>
>  *         <li>Beacon</li>
> ```

RF Beacon (iBeacon)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20)*
<!-- Sent from Reviewable.io -->

hippietilley committed:
MC-783 - Code review changes

hippietilley committed:
MC-783 - Testing for breakage when removing an empty constructor.

hippietilley committed:
MC-783 - Code review fixes

hippietilley committed:
MC-783 - Code review fixes

hippietilley commented:
Review status: 44 of 49 files reviewed at latest revision, 54 unresolved discussions.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java, line 55 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4bjH63kmRFGCs6KE6E:-L4l36J6sMsu_14oW7jn:b-l9b084) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/DataKitAPI.java#L55)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

Why 8KB?
</blockquote></details>

<< is a bitwise shift. So the value of BUFFER_SIZE ends up being 8192, or 8K. Why not just say BUFFER_SIZE = 8192? 

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java, line 110 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dF8BRQhDaOzoq1YIV:-L4lG8iWl4fElC3LAM13:b-671st4) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataType.java#L110)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

@monowar Is there a good reason for this?

@hippietilley Where else in the codebase is this called?  This may give an insight into this reason.  Give that this is an @override, there may be a reason.
</blockquote></details>

It's overriding an abstract method from Parcelable. It always 0 or 1 depending on whether the parcel has a file descriptor. Apparently none of our data types have file descriptors.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeBoolean.java, line 55 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dFchshYgbNmnVvVjF:-L4l11pKJEQnthc4S3Ng:b1v8kxe) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeBoolean.java#L55)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

Yes, assuming that it still compiles.
</blockquote></details>

I'll comment it out first and make sure the CI approves.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 84 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJU13cXq3oUhUXRA7:-L4l9Ak14PtuFAkfVis7:b-v5vxj7) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L84)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

While this is a typo, please revert it due to the potential breaking of the code.

@monowar Is this still used?
</blockquote></details>

Changed. Left the javadoc correctly spelled, added TODO to fix spelling so we can find it later.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 243 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dKX7U5T_lAb6Bq3mj:-L4lBzjwh1C2N4kl_VKW:b-w4bbtd) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L243)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

You messed up this string, check the original
</blockquote></details>

Yeah.... that makes more sense....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20)*
<!-- Sent from Reviewable.io -->

hippietilley committed:
MC-783 - Documented DataKitAPI and the callback interfaces

hippietilley committed:
MC-783 - First pass documentation for DataKitAPIExecute

hippietilley committed:
MC-783 - Reordered DataType methods to be consistent across types.

hippietilley committed:
MC-783 - Finished DataKitAPIExecute and proofread the rest of the code.

twhnat commented:
Reviewed 46 of 46 files at r3.
Review status: all files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeDouble.java, line 55 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4wFWPYcceWBf7k94Ha:-L4wFWPZh6Uvo6lv-N3X:b6isorg) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/datatype/DataTypeDouble.java#L55)):*
> ```Java
> 
>     /**
>      * TODO: Not used. Can remove?
> ```

Put this back in.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/AbstractObject.java, line 59 at r1](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4wFpug5kit_s9buyM4:-L4wFpug5kit_s9buyM5:b-6nvze6) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/e854d0bfaeeff1483c10ab0c34dc191dc21a5825/datakitapi/src/main/java/org/md2k/datakitapi/source/AbstractObject.java#L59)):*
> ```Java
>      * TODO: Does nothing. Can remove?
>      */
>     public AbstractObject(){
> ```

Put this back in

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 74 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4dJMgKFkHkrWHM1yNz:-L4wGOp8qcAV8HLhbxeO:b-i2dq97) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L74)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

Motion type inference
</blockquote></details>

Please use the correct word...

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 327 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4wGhv0KXU0y_cBcJyz:-L4wGhv0KXU0y_cBcJz-:b3ksp) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L327)):*
> ```Java
> 
>     /** Data quality summary for the day <p><code>"DATA_QUALITY_SUMMARY_DAY"</code></p> */
>     public static final String DATA_QUALITY_SUMMARY_DAY = "DATA_QUALITY_SUMMARY_DAY";
> ```

n

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20)*
<!-- Sent from Reviewable.io -->

hippietilley commented:
Review status: all files reviewed at latest revision, 17 unresolved discussions, some commit checks failed.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 327 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4wGhv0KXU0y_cBcJyz:-L59ffxwnZxJfgU6pqVc:b3kre) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L327)):*
<details><summary><i>Previously, twhnat (Timothy Hnat) wrote…</i></summary><blockquote>

n
</blockquote></details>

?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20)*
<!-- Sent from Reviewable.io -->

hippietilley committed:
MC-783 - Added dataType empty constructors back in and fixed typo

twhnat commented:
Reviewed 18 of 18 files at r4.
Review status: all files reviewed at latest revision, 14 unresolved discussions, some commit checks failed.

---

*[datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java, line 327 at r2](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20#-L4wGhv0KXU0y_cBcJyz:-L5KizvbqF6OcMvWQzXF:b-w41o83) ([raw file](https://github.com/md2korg/mcerebrum-datakitapi/blob/de66c2a06dd595758c14b3b1df46b5cd417ab03f/datakitapi/src/main/java/org/md2k/datakitapi/source/datasource/DataSourceType.java#L327)):*
<details><summary><i>Previously, hippietilley (Xavier Tilley) wrote…</i></summary><blockquote>

?
</blockquote></details>

Random character.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/md2korg/mcerebrum-datakitapi/20)*
<!-- Sent from Reviewable.io -->
